evennia/contrib/mail.py

```diff
@@ -0,0 +1,220 @@
+                    if "/" in self.rhs:
+                        message_number, message = self.rhs.split("/")
+                        if self.get_all_mail()[int(message_number) - 1]:
+                            old_message = self.get_all_mail()[int(message_number) - 1]
+
+                            self.send_mail(self.lhslist, "FWD: " + old_message.header,
+                                           message + "\n---- Original Message ----\n" + old_message.message,
+                                           self.caller)
+                            self.caller.msg("Message forwarded.")
+                        else:
+                            self.caller.msg("Message does not exist.")
+                            return
+                    else:
+                        if self.get_all_mail()[int(self.rhs) - 1]:
+                            old_message = self.get_all_mail()[int(self.rhs) - 1]
+                            self.send_mail(self.lhslist, "FWD: " + old_message.header,
+                                           "\n---- Original Message ----\n" + old_message.message, self.caller)
+                            self.caller.msg("Message forwarded.")
+                            old_message.tags.remove("u", category="mail")
+                            old_message.tags.add("f", category="mail")
+                        else:
+                            self.caller.msg("Message does not exist.")
+                            return
+            elif "reply" in self.switches:
+                if not self.rhs:
+                    self.caller.msg("You must define a message to reply to.")
+                    return
+                elif not self.lhs:
+                    self.caller.msg("You must supply a reply message")
+                    return
+                else:
+                    if self.get_all_mail()[int(self.lhs) - 1]:
+                        old_message = self.get_all_mail()[int(self.lhs) - 1]
+                        self.send_mail(old_message.senders, "RE: " + old_message.header,
+                                       self.rhs + "\n---- Original Message ----\n" + old_message.message, self.caller)
+                        old_message.tags.remove("u", category="mail")
+                        old_message.tags.add("r", category="mail")
+                        return
+                    else:
+                        self.caller.msg("Message does not exist.")
+                        return
+
+            else:
+                if self.rhs:
+                    if "/" in self.rhs:
+                        subject, body = self.rhs.split("/", 1)
+                    else:
+                        body = self.rhs
+                    self.send_mail(self.lhslist, subject, body, self.caller)
+                else:
+                    message = self.get_all_mail()[int(self.lhs) - 1]
+
+                    messageForm = []
+                    if message:
+                        messageForm.append(HEAD_CHAR * _WIDTH)
```

Comment:
HEAD_CHAR should be _HEAD_CHAR (several more examples in the file - best do a search & replace)